### ISSUE TITLE
Click on a color frame in color stack dump the stack

### ISSUE THREAD

**Issue by Alisu** [guillep/pharo-core#6690]
Describe the bug
Click on a color frame in color stack dump the stack

To Reproduce
1. In settings chose the option to color the stack in the debugger
2. Make a debugger open
3. Click on a color stack frame
4. All frames with other colors disappear

Expected behavior
I understand that you want to see all the frames in one color but i want to get back the others afterward.

Version information:
 - OS: [OSX Mojave]
 - Pharo 9

**Comment by dupriezt** [comment]
I did not manage to reproduce the bug.

Here is what I did:
I checked this setting 
![image](https://user-images.githubusercontent.com/32486709/85842717-6972c100-b7a0-11ea-8dac-435b4b6a07da.png)
Then executed a `self halt` in a playground.
A debugger opened like this:
![image](https://user-images.githubusercontent.com/32486709/85842815-90c98e00-b7a0-11ea-8f91-fc32dc56d5c7.png)
I clicked on the stack frame of the `halt` method like this:
![image](https://user-images.githubusercontent.com/32486709/85842906-b35ba700-b7a0-11ea-9620-e13e06b42a37.png)
But I did not see other frames disappear.

Pharo 9 - Ubuntu

**Comment by Alisu** [comment]
OK, i have it with the bug at #6403
<img width="1440" alt="Capture d’écran 2020-06-26 à 11 53 41" src="https://user-images.githubusercontent.com/17796500/85845015-df2c5c00-b7a3-11ea-9624-59260359cda8.png">
then click on do: and empty stack
<img width="1440" alt="Capture d’écran 2020-06-26 à 11 55 54" src="https://user-images.githubusercontent.com/17796500/85845100-04b96580-b7a4-11ea-9ec8-9503e6b60e3d.png">

**Comment by dupriezt** [comment]
Ok, I managed to reproduce it in that case.
So the steps to reproduce are:
- Set the "Color the stack" setting to true
- open a spotter
- type any letter
- ctrl/command + b (to browse classes)
- type any letter
- you get a DNU "Instance of ByteString did not understand #cull:"
- click the spotter window and close it with escape if it is still open
- in the debugger window, click on any stack frame
-> all the stack frame bu the first one (`doesNotUnderstand:`) disapear

[Edit]
I noticed that the problem does not appear (stack frames do not disappear in the debugger) for as long as the spotter window stays open.
When the "color the stack" setting is false, the problem does not appear even if the spotter window has been closed.

**Comment by dupriezt** [comment]
Hm. I said I didn't have the problem when the "color the stack" setting was false, but now I think it does not matter.
Sometimes, the stack still shows up in the debugger, but clicking on a stack frame does not inspect it.

I think the problem is that the spotter throws an exception, the debugger opens on it (and on the process in which the spotter was running), the spotter window gets closed, which triggers the termination of the corresponding process, which destroys the stack, which prevents the debugger from working normally.
And sometimes, the UI of the debugger will update to show the now empty stack, and other times it will not, but will still not let you inspect the stack frames it displays because it cannot access them anymore.

Tldr: it looks like the issue is that when the spotter window closes, it terminates the process in which it was running, even if that process was being debugged. And the debugger cannot work properly on a terminated process.

**Comment by StevenCostiou** [comment]
Any news on that issue @dupriezt @Alisu ?

**Comment by dupriezt** [comment]
I can't reproduce it in Pharo 9. Probably because the main debugger is the spec one now.

**Comment by StevenCostiou** [comment]
Ok thanks Thomas, I close it, reopen if it reproduces.
